Lib/tarfile.py

```diff
@@ -1164,8 +1165,8 @@ def _proc_gnulong(self, tarfile):
         # Fetch the next header and process it.
         try:
             next = self.fromtarfile(tarfile)
-        except HeaderError:
-            raise SubsequentHeaderError("missing or bad subsequent header")
+        except HeaderError as exc:
+            raise SubsequentHeaderError(str(exc)) from None
```

Comment:
Why is this change?

Reply:
`HeaderError` is generic (could be any of `EmptyHeaderError`, `TruncatedHeaderError`, `EOFHeaderError`, `InvalidHeaderError`, or even `SubsequentHeaderError`) and the specific error could be useful, so I'm saving the text from the specific error when converting to `SubsequentHeaderError`.

I do need to go back and change `as exc` to `as e` to match the rest of module.